apps/molgenis-components/src/components/forms/RowEdit.vue

```diff
@@ -163,40 +165,49 @@ export default {
         }
       });
     },
-    //create a filter in case inputs are linked by overlapping refs
-    refLinkFilter(column: IColumn) {
-      //need to figure out what refs overlap
-      if (
-        column.refLinkId &&
-        this.showColumn(column) &&
-        this.internalValues[column.refLinkId]
-      ) {
-        let filter: Record<string, any> = {};
-        this.tableMetaData.columns.forEach((column2: IColumn) => {
-          if (column2.id === column.refLinkId) {
-            this.schemaMetaData.tables.forEach((table: ITableMetaData) => {
-              //check how the refTableId overlaps with columns in our column
-              if (table.id === column.refTableId) {
-                table.columns.forEach((column3) => {
-                  if (
-                    column3.key === 1 &&
-                    column3.refTableId === column2.refTableId
-                  ) {
-                    filter[column3.id] = {
+    //update reflink filters and reset values if reflink changes
+    async updateRefLinks(changedColumn: IColumn) {
+      //need to figure out what refs overlap with the changed column
+      for (const column1 of this.tableMetaData.columns) {
+        if (
+          this.internalValues[changedColumn.id] &&
+          column1.refLinkId == changedColumn.id &&
+          this.showColumn(column1)
+        ) {
+          console.log("check");
+          //check how the refTableId overlaps with columns in our column
+          for (const table of this.schemaMetaData.tables) {
+            if (table.id === column1.refTableId) {
+              console.log("check2");
+              for (const column2 of table.columns) {
+                if (
+                  column2.key === 1 &&
+                  column2.refTableId === changedColumn.refTableId
+                ) {
```

Comment:
Since we are only interested in primary keys, wouldn't a find be more appropriate here to use a find instead of a for-loop?

Reply:
It would be be good to split out this loop and build a record with all the relevant keys and then combine that with column1, so we don't loop over the schemaMetadata several times. This would also reduce the complexity of the function a bit (it's kinda hard to read with so much nesting).

Reply:
that is a good idea. That would make it much more readible

Reply:
done